src/widget/wstemlabel.cpp

```diff
@@ -0,0 +1,51 @@
+#include "wstemlabel.h"
+
+#include "moc_wstemlabel.cpp"
+
+WStemLabel::WStemLabel(QWidget* pParent)
+        : WLabel(pParent),
+          m_stemInfo(QString(), QColor()),
+          m_stemNo(0) {
+}
+
+void WStemLabel::setup(const QDomNode& node, const SkinContext& context) {
+    m_stemNo = context.selectInt(node, "StemNum");
+}
+
+void WStemLabel::slotTrackUnloaded(TrackPointer track) {
+    Q_UNUSED(track);
+    m_stemInfo = StemInfo();
+    updateLabel();
+}
+
+void WStemLabel::slotTrackLoaded(TrackPointer track) {
+    if (!track) {
+        return;
+    }
+
+    auto stemInfo = track->getStemInfo();
+
+    if (stemInfo.isEmpty()) {
+        return;
+    }
+
+    m_stemInfo = stemInfo[m_stemNo - 1];
```

Comment:
assert that m_stemNo <= m_stemInfo.size()
otherwise we'd get `ASSERT failure in QList::operator[]: "index out of range"`